src/groups.rs

```diff
@@ -22,17 +26,37 @@ impl Groups {
         packages
     }
 
-    /// Return a new, empty Group
-    pub fn new() -> Self {
-        Self(BTreeMap::new())
-    }
-
     /// Loads Groups from a users pacdef config folder.
     ///
     /// # Errors
     ///  - If the Group config file cannot be found.
-    pub fn load() -> Result<Self> {
-        todo!()
+    pub fn load(group_dir: &Path) -> Result<Self> {
+        let mut groups = Self::default();
+
+        let group_dir = group_dir.join("groups/");
+        if !group_dir.is_dir() {
+            //other directories were already created with the config file
+            create_dir(&group_dir).context("group directory doesn't exist, creating one")?;
+        }
+
+        let mut files = vec![];
+        for file in WalkDir::new(&group_dir).follow_links(true) {
+            let path = file?.path().absolutize_from(&group_dir)?.to_path_buf();
+            files.push(path);
+        }
+        for group_file in files.iter().filter(|path| path.is_file()) {
+            let group_name = group_file
+                .strip_prefix(&group_dir)?
+                .to_str()
+                .ok_or(anyhow!("Will not fail on Linux"))?
+                .to_string();
+            println!("group_name: {group_name}");
+            let file_contents = read_to_string(group_file).context("Reading group file")?;
+
+            let packages: PackagesInstall = toml::from_str(&file_contents)?;
+            groups.insert(group_name, packages);
+        }
+        Ok(groups)
     }
```

Comment:
```suggestion
    /// Loads Groups from a users pacdef config folder.
    pub fn load(group_dir: &Path) -> Result<Self> {
        let mut groups = Self::default();

        let group_dir = group_dir.join("groups/");
        if !group_dir.is_dir() {
            return Err(anyhow!(
                "the groups directory was not found in the pacdef config folder, please create it"
            ));
        }

        let group_files: Vec<DirEntry> = WalkDir::new(&group_dir)
            .follow_links(true)
            .into_iter()
            .collect::<Result<_, _>>()?;

        for group_file in group_files.into_iter().filter(|x| x.path().is_file()) {
            let group_name = group_file
                .path()
                .strip_prefix(&group_dir)?
                .to_str()
                .context("converting group file name to a string")?
                .to_string();

            log::info!("parsing group file: {group_name}@{group_file:?}");

            let file_contents = read_to_string(group_file.path()).context("reading group file")?;

            let packages: PackagesInstall = toml::from_str(&file_contents)?;
            groups.insert(group_name, packages);
        }

        Ok(groups)
    }
```

It's probably best we get the user to change their config directory by creating a groups folder than attempting to modify it ourselves.